tls/s2n_connection.c

```diff
@@ -1168,6 +1170,9 @@ int s2n_connection_send_stuffer(struct s2n_stuffer *stuffer, struct s2n_connecti
 {
     notnull_check(conn);
     notnull_check(conn->send);
+    if (conn->write_fd_broken) {
+        S2N_ERROR(S2N_ERR_SEND_STUFFER_TO_CONN);
```

Comment:
Does S2N_ERR_IO or S2N_ERR_CLOSED make more sense here? When called from `s2n_flush`, only EAGAIN and EWOULDBLOCK are singled out, but S2N_ERR_IO is returned for all other conditions.

Reply:
Sure. S2N_ERR_CLOSED makes more sense although connection is half closed in this case. But as you mentioned, s2n_flush will return S2N_ERR_IO to the caller. Anyway, I will change the return code to S2N_ERR_CLOSED

Reply:
I'm guessing the S2N_ERR_CLOSED broken a bunch of integ tests / unit tests?

Reply:
> I'm guessing the S2N_ERR_CLOSED broken a bunch of integ tests / unit tests?

It should not as it will be converted to S2N_ERR_IO. However, the integration test did fail. I think the recv function also checks S2N_ERR_CLOSED so it might be OK to use S2N_ERR_SEND_STUFFER_TO_CONN as the caller will get  S2N_ERR_IO anyway.

Reply:
Sounds good. Not worth breaking the interface to add a new error code which ends up as S2N_ERR_IO anyway. Thanks!